cartographer/mapping_2d/local_trajectory_builder.cc

```diff
@@ -186,7 +186,7 @@ LocalTrajectoryBuilder::AddAccumulatedRangeData(
               {},  // 'low_resolution_point_cloud' is only used in 3D.
               {},  // 'rotational_scan_matcher_histogram' is only used in 3D.
               pose_estimate}),
-      std::move(insertion_submaps)});
+      std::move(insertion_submaps), gravity_aligned_range_data});
```

Comment:
The idea was that there is no need for two ways to expose this data, i.e. that this replaces `last_pose_estimate_`. Wouldn't it make sense to push the `scan_match_callback_` to the local trajectory builder and call it from here, i.e. no need to move the data out of local trajectory builder except through the call to the callback.

Reply:
As I mentioned in #508, my goal is to preserve the association of range data with the node ID (this was previously available because the range data was part of the trajectory node). We don't know the node ID yet in the local trajectory builder.

Reply:
I see. Then maybe keeping the `last_pose_estimate_` and replacing it by the callback in the global trajectory builder is the way to go. Then we would not have a `bool` but rather an optional `NodeId` depending on whether a node was inserted?

Reply:
How would you make the NodeId optional? By using a smart pointer (unique)?

Reply:
Yes, `std::unique_ptr<>` sounds good to me.

Reply:
Updated.